packages/indexer/src/sinks/drizzle/drizzle.test.ts

```diff
@@ -0,0 +1,207 @@
+      return generateMockMessages(5);
+    });
+
+    const sink = drizzleSink({ database: db });
+
+    const indexer = getMockIndexer({
+      sink,
+      override: {
+        transform: async ({ context, endCursor, block: { data } }) => {
+          const { transaction } = useSink({ context });
+
+          const db = transaction(endCursor);
+
+          // insert data for each message in db
+          await db
+            .insert(test_table)
+            .values([{ id: Number(endCursor?.orderKey), data }]);
+
+          // delete data for id 5000002 when orderKey is 5000004
+          // this is to test if the delete query is working
+          if (endCursor?.orderKey === 5000004n) {
+            await db.delete(test_table).where(eq(test_table.id, 5000002));
+          }
+        },
+      },
+    });
+
+    await run(client, indexer);
+
+    const result = await db.select().from(test_table);
+
+    expect(result).toHaveLength(5);
+
+    // as when you run delete query on a data, it isnt literally deleted from the db,
+    // instead, we just update the upper bound of that row to the current cursor
+    // check if the cursor upper bound has been set correctly
+    expect(
+      (
+        (result.find((r) => r.id === 5000002) as Record<string, unknown>)
+          ._cursor as Int8Range
+      ).range.upper,
+    ).toBe(5000004);
+  });
+
+  it("should select data", async () => {
+    const client = new MockClient<MockFilter, MockBlock>((request, options) => {
+      return generateMockMessages(5);
+    });
+
+    const sink = drizzleSink({ database: db });
+
+    let result: (typeof test_table.$inferSelect)[] = [];
+
+    const indexer = getMockIndexer({
+      sink,
+      override: {
+        transform: async ({ context, endCursor, block: { data } }) => {
+          const { transaction } = useSink({ context });
+
+          const db = transaction(endCursor);
+
+          // insert data for each message in db
+          await db
+            .insert(test_table)
+            .values([{ id: Number(endCursor?.orderKey), data }]);
+
+          // delete data for id 5000002 when orderKey is 5000004
+          // this will update the upper bound of the row with id 5000002 from infinity to 5000004
+          // so when we select all rows, row with id 5000002 will not be included
+          // as when we run select query it should only return rows with upper bound infinity
+          if (endCursor?.orderKey === 5000003n) {
+            await db.delete(test_table).where(eq(test_table.id, 5000002));
+          }
+
+          // when on last message of mock stream, select all rows from db
+          if (endCursor?.orderKey === 5000004n) {
+            result = await db.select().from(test_table);
+          }
+        },
+      },
+    });
+
+    await run(client, indexer);
+
+    expect(result).toHaveLength(4);
+    expect(result.find((r) => r.id === 5000002)).toBeUndefined();
```

Comment:
```suggestion
    expect(result.find((r) => r.id === 5000002)).toBeUndefined();
    
     const allRows = await db.select().from(test_table);
    expect(allRows).toHaveLength(5);
```

Let's add an extra check to make sure all data is still there. Technically already tested in delete, but that way it's clearer that we expect the indexer's view of the data to be different from what's in the db.